libsession/src/main/java/org/session/libsession/messaging/file_server/FileServerApi.kt

```diff
@@ -105,4 +110,52 @@ object FileServerApi {
         val request = Request(verb = HTTP.Verb.GET, endpoint = "file/$file")
         return send(request)
     }
+
+    /**
+     * Returns the current version of session
+     * This is effectively proxying (and caching) the response from the github release
+     * page.
+     *
+     * Note that the value is cached and can be up to 30 minutes out of date normally, and up to 24
+     * hours out of date if we cannot reach the Github API for some reason.
+     *
+     * https://github.com/oxen-io/session-file-server/blob/dev/doc/api.yaml#L119
+     */
+    suspend fun getClientVersion(): VersionData {
+        // Generate the auth signature
+        val secretKey =  MessagingModuleConfiguration.shared.getUserED25519KeyPair()?.secretKey?.asBytes
+            ?: throw (Error.NoEd25519KeyPair)
+
+        val blindedKeys = BlindKeyAPI.blindVersionKeyPair(secretKey)
+        val timestamp = TimeUnit.MILLISECONDS.toSeconds(System.currentTimeMillis()) //  The current timestamp in seconds
+        val signature = BlindKeyAPI.blindVersionSign(secretKey, timestamp)
+
+        // The hex encoded version-blinded public key with a 07 prefix
+        val blindedPkHex = buildString {
+            append("07")
+            append(blindedKeys.pubKey.toHexString())
+        }
+
+        val request = Request(
+            verb = HTTP.Verb.GET,
+            endpoint = "session_version",
+            queryParameters = mapOf("platform" to "android"),
+            headers = mapOf(
+                "X-FS-Pubkey" to blindedPkHex,
+                "X-FS-Timestamp" to timestamp.toString(),
+                "X-FS-Signature" to Base64.encodeToString(signature, Base64.NO_WRAP)
+            )
+        )
+
+        // transform the promise into a coroutine
+        val result = send(request).await()
+
+        // map out the result
+        val json = JsonUtil.fromJson(result, Map::class.java)
+        val statusCode = json.getOrDefault("status_code", 0) as Int
+        val version = json.getOrDefault("result", "") as String
+        val updated = json.getOrDefault("updated", 0.0) as Double
+
+        return VersionData(statusCode, version, updated)
```

Comment:
Seeing as there's all of these named vals that are simple enough to inline, they may as well be named params, and you get a little bit more safety that they're in the right order etc.

```suggestion
        return JsonUtil.fromJson(result, Map::class.java).let {
            VersionData(
                statusCode = it["status_code"] as? Int ?: 0
                version = it["result"] as? String ?: ""
                updated = it["updated"] as? Double ?: 0.0
            )
        }
```